src/vs/workbench/contrib/notebook/browser/contrib/notebookVariables/notebookVariablesView.ts

```diff
@@ -72,21 +72,50 @@ export class NotebookVariablesView extends ViewPane {
 			{
 				identityProvider: { getId: (e: INotebookVariableElement) => e.id },
 				horizontalScrolling: false,
-				supportDynamicHeights: true,
 				hideTwistiesOfChildlessElements: true,
 				accessibilityProvider: new NotebookVariableAccessibilityProvider(),
 				setRowLineHeight: false,
 			});
 
 		this.tree.layout();
 		this.tree?.setChildren(null, []);
+
+		this.tree.onDidChangeCollapseState(e => {
```

Comment:
try using an async tree instead of asynchronously updating the children when a node expands.